core/src/labelled.rs

```diff
@@ -617,6 +617,31 @@ where
     }
 }
 
+/// A generic representation for a `Result`.  This is not simply a `Result` to provide
+/// disambiguation between conversion to `Repr` and `transmogrification()` between `Repr`s.
+pub enum ResultRepr<T, E> {
+    Ok(T),
+    Err(E),
+}
+
+impl<T, E> LabelledGeneric for Result<T, E> {
+    type Repr = ResultRepr<Hlist!(Field<(), T>), Hlist!(Field<(), E>)>;
+
+    fn into(self) -> Self::Repr {
+        match self {
+            Ok(t) => ResultRepr::Ok(hlist!(field!((), t))),
+            Err(e) => ResultRepr::Err(hlist!(field!((), e))),
```

Comment:
Why an empty string for the field name? Is there precedent for this?

Reply:
I don't think so, this is (as far as I know) the first `Repr` for an `enum` type.  I considered using the variant name as the field name, but that doesn't translate well to an enum like:

```
enum Foo {
  Var1(Bar, Baz),
  Var2(Quux, Xyzzy)
}
```

Maybe it should agree with the naming for tuple structs (`_0`, `_1`...)?

Reply:
I would say go with `_0`.